README.md

```diff
@@ -1,38 +1,145 @@
+                    else
+                        return selectedFilterVal.some(
+                            (filterValue) => filterValue === workerFilterValue
+                        );
+                });
+            })
+            .map((filteredWorker) => {
+                return `${filteredWorker.imie} ${filteredWorker.nazwisko}`;
+            });
 
-## Installation
+        setFilteredWorkers(selected);
+    };
+    
+    
+    // ja bym to rozbił tak (trochę na szybko, więc niektóre rzeczy pominąłem)
+    const workerFilterValueMatch = () => {}
+    
+    const matchWorkersWithFilters = (defaultFilters, selectedFilters, workers) => {
+        return defaultFilters.every((filter) => {
+            const workerFilterValue = worker[filter];
+            const selectedFilterVal = selectedFilters[filter];
 
-Clone this repository and install dependencies inside project directory.
+            if (workerFilterValueMatch(workerFilterValue)) {
+                return selectedFilterVal.some((filterValue) =>
+                    workerFilterValue.includes(filterValue)
+                );
+            }
+            return selectedFilterVal.some(filterValue => filterValue === workerFilterValue);
+        });
+    }
+    
+    const getFilteredWorkers = (defaultFilters, selectedFilters, workers) => {
+        const selected = workers.reduce(
+            (array, worker) => {
+                const hasMatch = matchWorkersWithFilters(defaultFilters, selectedFilters, workers);
+                if (hasMatch) {
+                    return [
+                        ...array,
+                        `${worker.imie} ${worker.nazwisko}`
+                    ]
+                }
+                return array;
+            }, [])
+
+        setFilteredWorkers(selected);
+    };
 
-```
-git clone git@github.com:justFlow5/employee-filter-app.git
-cd employee-filter-app
-npm install
 ```
 
-## Usage
+Pytania:
+- Na przykładzie home albo text-field - poupychane są styled components? 
+- Czasem używane są do definiowania funkcji function a czasem () => , czy jest za tym jakaś logika
 
-To run this project in development mode use:
+- plik TextField.js 
 
+```javascript=
+    const [isActive, setIsActive] = useState('');
+    const [isFilled, setIsFilled] = useState('');
 ```
-npm start
+Czemu tutaj jest string zamiast np. boolean.
+Rozumiem, że niżej jest to używane jako klasa css , ale np clickedOutside jest już boolean i warunkowe wyświetlenie klasy?
+```jsx=
+ <FormField
+            className={`${isFilled} ${isActive} ${
+                isTimeInput ? 'timeType' : ''
+            } ${clickedOutside ? 'inactive' : ''} `}
+        >
 ```
 
----
-
-
-Created with [Create React App](https://github.com/facebook/create-react-app).
-Styled with [Styled-Components](https://github.com/styled-components/styled-components).
+- package.json
+Do czego używany jest jquery? Dlaczego @testing-library jest używane jako dependencies?
```

Comment:
jquery jest konieczny to działania tego jedno-inputowego kalendarza (react-bootstrap-daterangepicker). Prawdę mówiąc, miałem spore problemy ze znalezieniem biblioteki, która obsługiwałaby tylko jeden input, tak jak było to pokazane w specyfikacji zadania rekrutacyjnego. Wszystkie alternatywy były lżejsze i estetyczniejsze, ale miały dwa inputy. Tak więc ostatecznie przystałem na tę wersję, która jest rozwiązaniem dalekim od optymalnego. Byłem pewien, że zostanie mi wypomniane brak polskiego kalendarza, ale zmodyfikowanie go tak, by wyświetlał dane w języku polskim graniczyło z cudem ;)
@testing-library jest obecne w moich dependencies wyłącznie z powodu mojej nieuwagi. Tworząc aplikację przy użyciu create-react-app jest on instalowany automatycznie w dependiencies i zapomniałem go usunąć. Co prawda nie wpływa to w żaden sposób na rozmiar bundle'a, więc szkód jako takich nie ma to rozumiem oczywiście, że nieużywane dependencies powinny być oczywiście usunięte z package.json - mój błąd.

Reply:
👌